### PR TITLE
[ENH] Do sklearn checks on SurfaceLabelsMasker

### PR DESCRIPTION
**Pull request by man-shu**
- Closes None

Changes proposed in this pull request:

- sklearn checks were happening on SurfaceMasker instead of SurfaceLabelsMasker

### CODE REVIEW

File: nilearn/maskers/surface_labels_masker.py [hunk 1]
```diff
@@ -95,7 +95,7 @@ class SurfaceLabelsMasker(TransformerMixin, CacheMixin, BaseEstimator):
 
     def __init__(
         self,
-        labels_img,
+        labels_img=None,
```
Comment (man-shu) [inline nilearn/maskers/surface_labels_masker.py:98]: So this seems to pass the `check_parameters_default_constructible` test, but I am not quite sure what complies the best with sklearn guidelines i.e.

whether to have `labels_img` as a parameter to `.fit()` or maintain our old behavior of letting users pass it during instantiation (but now it won't throw any error if they don't)

Comment (man-shu) [reply nilearn/maskers/surface_labels_masker.py:98]: Guidelines here: https://scikit-learn.org/stable/developers/develop.html#instantiation

Comment (Remi-Gau) [reply nilearn/maskers/surface_labels_masker.py:98]: If we do this we must thrown an error at fit if labels_img is None.

I think it'd be good to do this for all maskers: looking at NiftiMasker (which I believe is the oldest and most canonical) it allows None for the maask image.

File: nilearn/maskers/tests/test_surface_labels_masker.py [hunk 2]
```diff
@@ -392,10 +393,6 @@ def test_surface_label_masker_inverse_transform_list_surf_images(
     assert img.shape == (surf_label_img.mesh.n_vertices, 7)
 
 
-def test_surface_label_masker_list_inverse_transform_output(surf_mesh):
```
Comment (man-shu) [inline nilearn/maskers/tests/test_surface_labels_masker.py:395]: unrelated: seems like I started defining a test in a previous branch but then refactored other tests to test the same thing

File: nilearn/maskers/surface_labels_masker.py (hunk 1 above)
Comment (bthirion) [reply nilearn/maskers/surface_labels_masker.py:98]: It's because the mask image can be computed from the data when using `NiftiMasker`.
To me it's OK to provide `labels_img` at instantiation time rather than fit time.

**Review by bthirion — COMMENTED** [review]
LGTM.

File: nilearn/maskers/surface_labels_masker.py [hunk 3]
```diff
@@ -168,6 +168,12 @@ def fit(self, img=None, y=None):
         """
         del img, y
 
+        if self.labels_img is None:
+            raise ValueError(
+                "Please provide a labels_img to the masker. For example, "
+                "masker = SurfaceLabelsMasker(labels_img=labels_img)"
+            )
```
Comment (Remi-Gau) [inline nilearn/maskers/surface_labels_masker.py:175]: Will probably need a test.

Comment (man-shu) [reply nilearn/maskers/surface_labels_masker.py:162]: Done!

File: nilearn/maskers/tests/test_surface_labels_masker.py [hunk 4]
```diff
@@ -6,9 +6,22 @@
 
 from nilearn._utils.class_inspect import check_estimator
 from nilearn._utils.helpers import is_matplotlib_installed
-from nilearn.maskers import SurfaceLabelsMasker, SurfaceMasker
+from nilearn.maskers import SurfaceLabelsMasker
 from nilearn.surface import SurfaceImage
 
+
+@pytest.fixture
+def _sklearn_surf_label_img(surf_mesh):
```
Comment (man-shu) [inline nilearn/maskers/tests/test_surface_labels_masker.py:14]: Had to create a new label img fixture locally because an invalid scikit-learn check was passing for unrelated reasons...

**Review by bthirion — APPROVED** [review]
LGTM, thx.

File: nilearn/reporting/data/html/report_body_template.html [hunk 5]
```diff
@@ -24,11 +24,6 @@
       {{endif}}
     </div>
   </div>
-  <div class="pure-u-1 pure-u-md-1-3">
-    {{if description}}
-    <p>{{description}}</p>
-    {{endif}}
-  </div>
```
Comment (man-shu) [inline nilearn/reporting/data/html/report_body_template.html:31]: If we don't define `description` during instantiation, we can't expect it here

Comment (man-shu) [reply nilearn/reporting/data/html/report_body_template.html:31]: But it should nevertheless show up after fitting because the `report_body_template_surfacemasker.html` template has a field for it.

Actually let me confirm that

Comment (man-shu) [reply nilearn/reporting/data/html/report_body_template.html:31]: Yes it does 👍🏽

**Comment by man-shu** [comment]
All tests are passing, so now it's ok to review @Remi-Gau

**Review by Remi-Gau — APPROVED** [review]

**Comment by Remi-Gau** [comment]
thanks @man-shu

**Comment by man-shu** [comment]
Thanks for the reviews!
